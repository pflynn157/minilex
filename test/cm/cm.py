base_path = "./base"
output_path = "./src"

# Define the keywords
keywords = [
    ("func", "Func"), ("is", "Is"), ("end", "End"),
    ("var", "Var"), ("return", "Return")
]

# Define the symbols
symbols = [
    (";", "SemiColon"), ("=", "Assign"), (":", "Colon"), (":=", "Assign2")
]

# Define single-line comments
single_comments = [ "#" ]

# Define multi-line comments
multi_comments = [ ("/*", "*/"), ("/+", "."), ("/-", "---/") ]
