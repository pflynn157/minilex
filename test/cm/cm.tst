
func main is
    /+
    var x = 20;
    .
    
    var y = "Hello!";
    
    /-
    var z = x;
    ---/
    
    /*
    var xx : int := 32;
    return z;
    */
    # This is a comment
end


