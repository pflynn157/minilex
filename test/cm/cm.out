func
ID(main)
is
var
ID(y)
=
STR(Hello!)
;
end

